src/components/App/App.js

```diff
@@ -0,0 +1,169 @@
+import { Routes, Route } from "react-router-dom";
+import SignUpPopup from "../SignUpPopup/SignUpPopup";
+import SignInPopup from "../SignInPopup/SignInPopup";
+import MessagePopup from "../MessagePopup/MessagePopup";
+import ProtectedRoute from "../ProtectedRoute/ProtectedRoute";
+import { useEffect, useState } from "react";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
+import Footer from "../Footer/Footer";
+import Main from "../Main/Main";
+import mainApi from "../../utils/MainApi";
+import * as auth from "../../utils/auth";
+import SavedNews from "../SavedNews/SavedNews";
+
+import "./App.css";
+
+function App() {
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [currentUser, setCurrentUser] = useState({
+    userName: "",
+    email: "",
+    owner:""
+  });
+  const [isSignInPopupOpen, setIsSignInPopupOpen] = useState(false);
+  const [isSignUpPopupOpen, setIsSignUpPopupOpen] = useState(false);
+  const [isMsgPopupOpen, setIsMsgPopupOpen] = useState(false);
+  const [articleData, setArticleData] = useState({});
+  const [articleDbData, setArticleDbData] = useState({});
+  const [search, setSearch] = useState("");
+
+  const handleLogin = (email, name, owner) => {
+    setLoggedIn(true);
+    setCurrentUser({ email, userName: name, owner:owner });
+  };
+
+  const handleLogout = () => {
+    setLoggedIn(false);
+    setCurrentUser({
+      email: "",
+      userName: "",
+      owner:""
+    });
+    localStorage.removeItem("token");
+    return true;
+  };
+
+  function anyPopUpOpen() {
+    const popups = [isSignInPopupOpen, isSignUpPopupOpen, isMsgPopupOpen];
+    const areOpen = (element) => element;
+    return popups.some(areOpen) ? true : false;
+  }
+
+  function closeAllPopups() {
+    setIsSignInPopupOpen(false);
+    setIsSignUpPopupOpen(false);
+    setIsMsgPopupOpen(false);
+  }
+
+  function getArticles(){
+    loggedIn &&
+      mainApi.getSavedArticles().then((articles) => {
+        const userSavedArticles = articles.filter(article => article.owner === currentUser.owner);
+        setArticleDbData(userSavedArticles);
+      });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**